src/SonarScanner.MSBuild.PreProcessor/Interfaces/IPreprocessorObjectFactory.cs

```diff
@@ -30,7 +32,7 @@ public interface IPreprocessorObjectFactory
         /// </summary>
         /// <param name="args">Validated arguments.</param>
         /// <remarks>It is the responsibility of the caller to dispose of the server, if necessary.</remarks>
-        ISonarWebService CreateSonarWebService(ProcessedArgs args);
+        Task<ISonarWebService> CreateSonarWebService(ProcessedArgs args, IDownloader downloader = null);
```

Comment:
The downloader parameter is needed for testing purposes, so we can mock the web API calls.

Reply:
I don't like it, and that's all I can do about it 😆 😅 I don't see a nicer way